### PR TITLE
initial dev

### PR DESCRIPTION
**Pull request by Nir0n**

### CODE REVIEW

File: category.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        if isinstance(obj, UUID):
+            return str(obj)
+        if isinstance(obj, datetime.datetime):
+            return str(obj)
+        return super().default(obj)
+
+
+@router.get("/categories", response_model=List[CategoryOut])
+async def get_categories():
+    return session.get("/categories")
```
Comment (stretchcom) [inline category.py]: base_url ?

File: client.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            coord=[latitude, longitude],
+            country=country,
+            city=city,
+            state=state,
+            line1=line1,
+            line2=line2,
+            zip=zip
+        )
+    )
+    res = session.post(f"{base_url}/client", data=json.dumps(dto_in.dict()))
```
Comment (stretchcom) [inline client.py]: We can use  json=dto_in.dict().  instead convertion

File: search.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+class SearchOut(PaginationIn, BaseModel):
+    profiles: list[VendorShortOut] = Field(default=[])
+
+# -------------------------------------------------------------------------------------------------
+
+
+router = APIRouter()
+session = requests.Session()
+base_url = os.getenv('STRETCH_PUBLIC_API_URL', 'http://localhost:8000/api/v1/public')
```
Comment (stretchcom) [inline search.py]: default value better setup on api.stretch.com

File: search.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+            return str(obj)
+        if isinstance(obj, datetime.datetime):
+            return str(obj)
+        return super().default(obj)
+
+
+
+@router.post("/search", response_model=SearchOut)
+async def search(dto_in: SearchIn):
+    return  session.post("/search", data=json.dumps(dto_in.dict(), cls=JSONEncoder))
```
Comment (stretchcom) [inline search.py]: We return json ?

**Review by stretchcom — CHANGES_REQUESTED** [review]

File: search.py (hunk 4 above)
Comment (Nir0n) [reply search.py]: yes

**Review by stretchcom — APPROVED** [review]

**Pull request merged**